moi.js

```diff
@@ -57,41 +57,56 @@ var moi = new function () {
     this.send = function(action, data) {
         to_send = {}; 
         to_send[action] = data;
-        ws.send(encode(to_send))
+        ws.send(encode(to_send));
     };
-    
+
     /**
      *
      * Verify the browser supports websockets, and if so, initialize the
      * websocket. On construction, this method will send a message over the
      * socket to get all known job information associated with this client.
      *
-     * @param {group_id} A group ID to get initial data from, or null to fetch 
-     * all records associated with the user. 
+     * @param {host} The URL for the websocket, minus the ws:// header, or null
+     * to use the default moi-ws.
+     * @param {group_id} A group ID to get initial data from, or null to fetch
+     * all records associated with the user.
+     * @param {on_open} Optional function for action when websocket is opened.
+     * @param {on_close} Optional function for action when websocket is closed.
+     * @param {on_error} Optional function for action when websocket errors.
      */
-    this.init = function(group_id) {
+    this.init = function(host, group_id, on_open, on_close, on_error) {
+        host = host || window.location.host + '/moi-ws/';
         if (!("WebSocket" in window)) {
             alert("Your browser does not appear to support websockets!");
             return;
         }
-        ws = new WebSocket(host);
-        
+        //check if we need regular or secure websocket
+        socket = 'ws://';
```

Comment:
ternary operator. and it has its c-style syntax. so awesome

On Mon, Apr 27, 2015 at 11:26 AM, Joshua Shorenstein <
notifications@github.com> wrote:

> In moi.js
> <https://github.com/biocore/mustached-octo-ironman/pull/19#discussion_r29168867>
> :
>
> >          if (!("WebSocket" in window)) {
> >              alert("Your browser does not appear to support websockets!");
> >              return;
> >          }
> > -        ws = new WebSocket(host);
> > -
> > +        //check if we need regular or secure websocket
> > +        socket = 'ws://';
>
> Didn't know that was a thing. Cool!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/mustached-octo-ironman/pull/19/files#r29168867>
> .
>